common/environment/build-style/cmake.sh

```diff
@@ -1,5 +1,7 @@
 if [ "$CHROOT_READY" ]; then
-	hostmakedepends+=" cmake"
+	if [ "$pkgname" != cmake ]; then
+		hostmakedepends+=" cmake"
+	fi
```

Comment:
This feels a bit too magical, but idk what others think. Arguably it's because adding `cmake` to hostmakedepends happens automagically already and we end up needing to work around it.

At least having a variable to skip it would be less magical, though also complex.

Reply:
I would say, let's remove those `hostmakedepends` in `environment`, force template add them explicitly.